pkg/controller/machine.go

```diff
@@ -354,6 +354,13 @@ func (c *controller) machineCreate(machine *v1alpha1.Machine, driver driver.Driv
 			LastUpdateTime: metav1.Now(),
 		}
 		c.updateMachineStatus(machine, lastOperation, currentStatus)
+
+		// make sure that the failed machine doesn't exist
+		if e := driver.Delete(); e != nil {
```

Comment:
IMO I don't think we should be doing this here. I think this should be handled at the driver to trigger such deletion. Because all the MCM should care is if creation went through or not and if not error out. Creation logic at MCM shouldn't care about delete on failure. I think it can be better handled at the driver itself. Something like this - https://github.com/gardener/machine-controller-manager/blob/master/pkg/driver/driver_azure.go#L408-L413

Reply:
Fixed